### PR TITLE
Pin modes parser

### PR DESCRIPTION
**Pull request by yrabbit**
A simple parser of pin mode flags. Only understands one flag per line.

### CODE REVIEW

**Comment by pepijndevos** [comment]
(This is part of https://github.com/YosysHQ/apicula/pull/35 )

File: gowin/arch.cc
Comment (gatecat) [inline gowin/arch.cc:523]: if you are setting a boolean attribute, setting it to 1 is more appropriate than 0

File: gowin/arch.cc
Comment (gatecat) [inline gowin/arch.cc:512]: this could do with a `make clangformat` run to normalise the formatting

File: gowin/arch.cc
Comment (yrabbit) [reply gowin/arch.cc:523]: I'm only interested in having the attribute itself, so I'll correct it to 1.

File: gowin/arch.cc
Comment (yrabbit) [reply gowin/arch.cc]: I found clang-format10.

File: gowin/arch.cc
Comment (gatecat) [reply gowin/arch.cc]: seems like your last commit fixing the formatting accidentally set this back to 0

Comment (yrabbit) [reply gowin/arch.cc]: Damn. I need to be more careful.

**Review by gatecat — APPROVED** [review]
LGTM. @pepijndevos, would you like this to wait until the apicula side is merged?

**Comment by pepijndevos** [comment]
I don't think it depends on Apicula, just that it will not work if you use these flags. So probably fine to merge.
I'll probably review the Apicula side this weekend.

**Pull request merged**
